sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -363,21 +363,16 @@ export class MessageSession extends LinkEntity {
         let errorMessage: string = "";
         // SB allows a sessionId with empty string value :)
 
-        if (receivedSessionId == null) {
-          if (this.sessionId == null) {
-            errorMessage = `No unlocked sessions were available`;
-          } else {
-            errorMessage =
-              `Received an incorrect sessionId '${receivedSessionId}' while creating ` +
-              `the receiver '${this.name}'.`;
-          }
-        }
-
-        if (this.sessionId != null && receivedSessionId !== this.sessionId) {
-          errorMessage =
-            `Received sessionId '${receivedSessionId}' does not match the provided ` +
-            `sessionId '${this.sessionId}' while creating the receiver '${this.name}'.`;
+        if (this.sessionId == null && receivedSessionId == null) {
+          // Ideally this code path should never be reached as `createReceiver()` should fail instead 
+          // TODO: https://github.com/Azure/azure-sdk-for-js/issues/9775 to figure out why this code path indeed gets hit.
+          errorMessage = `No unlocked sessions were available`;
+        } else if (this.sessionId != null && receivedSessionId !== this.sessionId) {
+          // This code path is reached if the session is already locked by another receiver.
+          // TODO: Check why the service would not throw an error or just timeout instead of giving a misleading successful receiver
```

Comment:
Did this happen? or is it just an additional guard?

Reply:
I'm actually curious if we have tests for this particular part. Do you know @ramya-rao-a?

Reply:
Yes, added one in this PR

Reply:
@HarshaNalluru When a session is already locked and we ask for another receiver to the same session,  `receivedSessionId` here would be `undefined`  and so we will get to the else if block here. You can see the test case I added for it.

AFAIK, we have not seen a case where `receivedSessionId` is a non null/undefined value and is different from given `sessionId`. But the check for it existed before, so I kept it.